tests/components/agent_dvr/test_alarm_control_panel.py

```diff
@@ -0,0 +1,31 @@
+"""Tests for the Agent DVR alarm control panel."""
+from typing import Optional
+from unittest.mock import AsyncMock, Mock, patch
+
+from homeassistant.components.agent_dvr.alarm_control_panel import AgentBaseStation
+
+
+def client_mock(profile: Optional[str] = "home"):
+    """Use this to mock the agent client."""
+    api = Mock()
+    api.update = AsyncMock()
+    api.get_active_profile = AsyncMock(return_value=profile)
+    return api
+
+
+async def test_update():
+    """Test update happy path."""
+    with patch("homeassistant.components.agent_dvr"):
+        api = client_mock()
+        alarm = AgentBaseStation(api)
```

Comment:
Please set up the integration while patching appropriately and let the platform add an entity. Then assert the entity state in the core state machine. Don't access any integration details in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations